File: python/program/Elektronik.py
class Elektronik:
    """
    Sebuah class untuk merepresentasikan produk elektronik
    menggunakan metode getter dan setter konvensional.
    """

    def __init__(
        self,
        id_produk: str = "",
        nama_produk: str = "",
        harga_produk: int = 0,
        stok_produk: int = 0,
        foto_produk: str = "",
    ):
        """Constructor untuk class Elektronik."""
        self.id_produk = id_produk
        self.nama_produk = nama_produk
        self.harga_produk = harga_produk
        self.stok_produk = stok_produk
        self.foto_produk = foto_produk

    # --- Getters ---
    def get_id_produk(self) -> str:
        return self.id_produk

    def get_nama_produk(self) -> str:
        return self.nama_produk

    def get_harga_produk(self) -> int:
        return self.harga_produk

    def get_stok_produk(self) -> int:
        return self.stok_produk

    def get_foto_produk(self) -> str:
        return self.foto_produk

    # --- Setters ---
    def set_id_produk(self, value: str):
        self.id_produk = value

    def set_nama_produk(self, value: str):
        self.nama_produk = value

    def set_harga_produk(self, value: int):
        if value < 0:
            print("Warning: Harga produk tidak boleh negatif.")
        self.harga_produk = value

    def set_stok_produk(self, value: int):
        if value < 0:
            print("Warning: Stok produk tidak boleh negatif.")
        self.stok_produk = value

    def set_foto_produk(self, value: str):
        self.foto_produk = value

File: cpp/program/Elektronik.cpp
#include <iostream>
#include <string>

using namespace std;

class Elektronik {
private:
  string id_produk, nama_produk, foto_produk; 
  int harga_produk, stok_produk;

public:
  Elektronik() {
    this->id_produk = "";
    this->nama_produk = "";
    this->harga_produk = 0;
    this->stok_produk = 0;
    this->foto_produk = "";
  }

  Elektronik(string id_produk, string nama_produk, int harga_produk, int stok_produk, string foto_produk) {
    this->id_produk = id_produk;
    this->nama_produk = nama_produk;
    this->harga_produk = harga_produk;
    this->stok_produk = stok_produk;
    this->foto_produk = foto_produk;
  }

  // --- GETTERS ---
  // Metode untuk mengambil nilai atribut

  string getIdProduk() {
      return id_produk;
  }

  string getNamaProduk() {
      return nama_produk;
  }

  int getHargaProduk() {
      return harga_produk;
  }

  int getStokProduk() {
      return stok_produk;
  }

  string getFotoProduk() {
      return foto_produk;
  }

  // --- SETTERS ---
  // Metode untuk mengubah nilai atribut

  void setIdProduk(string id_produk) {
      this->id_produk = id_produk;
  }

  void setNamaProduk(string nama_produk) {
      this->nama_produk = nama_produk;
  }

  void setHargaProduk(int harga_produk) {
      this->harga_produk = harga_produk;
  }

  void setStokProduk(int stok_produk) {
      this->stok_produk = stok_produk;
  }

  void setFotoProduk(string foto_produk) {
      this->foto_produk = foto_produk;
  }

  ~Elektronik(){}
};
